test/jdk/java/util/regex/NamedGroupsTests.java

```diff
@@ -346,4 +348,24 @@ private static void testPatternNamedGroupsTwoNamedGroups() {
         }
     }
 
+    private static void testMatchAfterUsePattern() {
+        Pattern p1 = Pattern.compile("(?<a>...)(?<b>...)");
+        Matcher m = p1.matcher("foobar");
+        if (!m.matches()) {
+            throw new RuntimeException();
+        }
+        if (!m.group("a").equals("foo")) {
+            throw new RuntimeException();
+        }
+
+        Pattern p2 = Pattern.compile("(?<b>...)(?<a>...)");
+        m.usePattern(p2);
+        if (!m.matches()) {
+            throw new RuntimeException();
+        }
+        if (!m.group("a").equals("bar")) {
+            throw new RuntimeException();
```

Comment:
I agree.
On the one hand, I would prefer to rewrite these tests (the .java file) JUnit tests, but this would require more review work.
On the other hand, adding descriptions just for this specific, additional test is a bit at odd with the other tests in the file.
What's the preferred course of action? I have no problems with either way.

Reply:
I'd add string descriptions to the new exceptions and file an issue to update the test later.